feat(docs): Clarify gatsby-plugin-mdx requirements for gatsby-source-filesystem (#15604)
gatsbyjs/gatsby: packages/gatsby-plugin-mdx/README.md

```js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`,
      },
    },
    `gatsby-plugin-mdx`,
  ],
}
```

By default, this configuration will allow you to create pages
with `.mdx` files in `src/pages` and will process any Gatsby nodes
with Markdown media types into MDX content.

> Note that gatsby-mdx requires gatsby-source-filesystem to be present

Reviewer: ```suggestion
Note that gatsby-plugin-mdx requires gatsby-source-filesystem to be present
```

> and configured to process your markdown files to correctly

Reviewer: ```suggestion
and configured to process local markdown files in order to
```

// gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-filesystem`,
      options: {

>         name: `pages`,

Reviewer: Pages should be automatically sourced with MDX if there are any `.mdx` files. Were you not seeing this occur?

Author: Page were sourced, yes. But graphql node were not created for the pages, that's why I found it quite confusing, and that's why I think it's a good idea to include filesystem plugin in examples
